dbms/src/Storages/MergeTree/checkDataPart.cpp

```diff
@@ -379,7 +381,8 @@ MergeTreeData::DataPart::Checksums checkDataPart(
             size_t read_size = tmp_column->size();
             column_size += read_size;
 
-            if (read_size < rows_after_mark || mark_num == adaptive_index_granularity.getMarksCount())
```

Comment:
Why do we remove condition `read_size < rows_after_mark`? Isn't it necessary for non-adaptive index granularity?

Reply:
No, in non-adaptive table we must read as much marks as in `adaptive_index_granularity.getMarksCountWithoutFinal`.

Reply:
But I'll add a test for it.

Reply:
Can you please also add test with number of rows  != 8192 * N (for non-adaptive table)?